code/code/disc/disc_hand_of_god.cc

```diff
@@ -891,27 +891,26 @@ int portal(TBeing * caster, const char * portalroom, int level, short bKnown)
 
 
   if (caster->bSuccess(bKnown,caster->getPerc(),SPELL_PORTAL)) {
-    TPortal * tmp_obj = new TPortal(rp);
-    tmp_obj->setPortalNumCharges(1*((level/5)+1));
-
     if (critSuccess(caster, SPELL_PORTAL)) {
       CS(SPELL_PORTAL);
-      tmp_obj->obj_flags.decay_time *= 2;
-      tmp_obj->setPortalNumCharges(2*((level/5)+1));
+      num_charges *= 2;
     }
+
+    TPortal * tmp_obj = new TPortal(rp);
+    tmp_obj->setPortalNumCharges(num_charges);
     *caster->roomp += *tmp_obj;
 
     if (tPerson)
       tmp_obj->checkOwnersList(tPerson);
 
-    caster->roomp->playsound(SOUND_SPELL_PORTAL, SOUND_TYPE_MAGIC);
-
     TPortal * next_tmp_obj = new TPortal(caster->roomp);
+    next_tmp_obj->setPortalNumCharges(num_charges);
     *rp += *next_tmp_obj;
 
     if (tPerson)
       next_tmp_obj->checkOwnersList(tPerson);
 
+    caster->roomp->playsound(SOUND_SPELL_PORTAL, SOUND_TYPE_MAGIC);
```

Comment:
I wonder if we should take the trouble to add MSP support for webclient?

Reply:
I've never used sound and wouldn't use it. Seems like it would be a cool thing to add into the web client though

Reply:
I haven't used sound since the old Brutius client that came with a sound pack.

In this day and age there are TONS of free music and sound effect pack assets that come with the myriad of game-making programs. It probably wouldn't be very difficult to put something together.

It certainly wouldn't hurt to build towards the eventuality of having some sound effects in the web client.